python/pyspark/sql/types.py

```diff
@@ -410,6 +410,24 @@ def __init__(self, name, dataType, nullable=True, metadata=None):
         self.dataType = dataType
         self.nullable = nullable
         self.metadata = metadata or {}
+        self.needConversion = dataType.needConversion
+        self.toInternal = dataType.toInternal
+        self.fromInternal = dataType.fromInternal
+
+    def __getstate__(self):
```

Comment:
We need to handle pickle by ourselves because we have fields with function values